### PR TITLE
update security recommendations

### PR DESCRIPTION
**Pull request by screeley44**
@jeffvance @copejon @erinboyd

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -152,3 +152,54 @@ This is the template PVC. No namespace is supplied since the PVC is excpected to
 1. Monitor the importer pod:
 
         $ kubectl logs <unique-name-of-importer pod>  # shown in controller log above
+
+### Security Configurations
```
Comment (jeffvance) [inline README.md]: I'd like to see this section listed in the ToC

File: README.md [hunk 2]
```diff
@@ -152,3 +152,54 @@ This is the template PVC. No namespace is supplied since the PVC is excpected to
...
+
+#### RBAC Roles
+
+CDI needs certain permissions to be able to execute properly, primarily the `cluster-admin` role should be applied to the service account being used through the [Kubernetes RBAC model](https://kubernetes.io/docs/admin/authorization/rbac/).  For example, if CDI is running in a namespace called `golden-images`, and the `default` service account is being used, then the following RBAC should be applied:
+
+```
+  $ kubectl create clusterrolebinding <binding name> --clusterrole=cluster-admin  --serviceaccount=<namespace>:default
+
+  i.e.
```
Comment (jeffvance) [inline README.md]: +1 showing a specific example of the general cmd!

File: README.md [hunk 3]
```diff
@@ -152,3 +152,54 @@ This is the template PVC. No namespace is supplied since the PVC is excpected to
...
+  $ kubectl create clusterrolebinding c-golden-images-default --clusterrole=cluster-admin  --serviceaccount=golden-images:default
+
+```
+
+> NOTE: This gives full cluster-admin access to this binding
```
Comment (jeffvance) [inline README.md]: Has there been any work done to determine the minimal privileges/role needed for the controller to be able to list pvcs in its own namespace?

File: README.md [hunk 4]
```diff
@@ -152,3 +152,54 @@ This is the template PVC. No namespace is supplied since the PVC is excpected to
...
+- Lock Down StorageClass Usage for Namespace:
+
+```
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  name: protect-mynamespace
+spec:
+  hard:
+    <storageclass name>.storageclass.storage.k8s.io/requests.storage: "0"
```
Comment (jeffvance) [inline README.md]: nit: I'd put a dash between _<storageclass_ and _name>_ here, or combine them as _<storageClassName>_. The idea being to not have a space in this token.

File: README.md (hunk 3 above)
Comment (screeley44) [reply README.md]: There has been some investigation, but think that will need to be fine tuned some more, but initially we wanted the doc to reflect our ansible role, as it moves forward, we will update the doc as well

File: README.md (hunk 4 above)
Comment (screeley44) [reply README.md]: ack

File: README.md [hunk 5]
```diff
@@ -152,3 +152,54 @@ This is the template PVC. No namespace is supplied since the PVC is excpected to
...
+- Open Up StorageClas Usage for Namespace:
+
+```
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  name: protect-mynamespace
+spec:
+  hard:
+    <storageclass name>.storageclass.storage.k8s.io/requests.storage: "500Gi"
```
Comment (jeffvance) [inline README.md]: ditto re. space in placeholder value.

File: README.md [hunk 6]
```diff
@@ -152,3 +152,54 @@ This is the template PVC. No namespace is supplied since the PVC is excpected to
...
+  name: protect-mynamespace
+spec:
+  hard:
+    <storageclass name>.storageclass.storage.k8s.io/requests.storage: "0"
+```  
+
+> NOTE: <storage-class-name>.storageclass.storage.k8s.io/persistentvolumeclaims: "0" would also accomplish the same affect by not allowing any pvc requests against the storageclass for this namespace.
+
+
+- Open Up StorageClas Usage for Namespace:
```
Comment (jeffvance) [inline README.md]: typo: StorageClas**s**

Comment (screeley44) [reply README.md]: ack

File: README.md (hunk 5 above)
Comment (screeley44) [reply README.md]: ack

File: README.md (hunk 1 above)
Comment (screeley44) [reply README.md]: ack

**Comment by erinboyd** [comment]
LGTM

**Comment by jeffvance** [comment]
waiting for @copejon to merge since he's working on the readme me too.

**Pull request merged**